src/app/CommandHandler.h

```diff
@@ -50,8 +53,29 @@ class CommandHandler : public Command
 
 private:
     friend class TestCommandInteraction;
+    friend class CommandHandlerHandleDeleter;
     CHIP_ERROR SendCommandResponse();
     CHIP_ERROR ProcessCommandDataElement(CommandDataElement::Parser & aCommandElement) override;
+
+    struct CommandHandlerHandleDeleter
+    {
+        void operator()(CommandHandler * apCommandHandler) const;
+    };
+
+    uint32_t mPendingWorkCount = 0;
+
+public:
+    using CommandHandlerAsyncHandle = std::unique_ptr<CommandHandler, CommandHandlerHandleDeleter>;
+    /**
+     * @brief Usually, the command handler do things very fast, and a synchronous handler is OK, however, some command handler might
+     * require some background work, thus we need to mark the current CommandHander as a async command handler.
+     * Releasing the CommandHandlerAsyncHandle means the owner has finished the work and put necessary responses, and the response
+     * can be sent.
+     *
```

Comment:
Thanks for adding a comment to explain the function, we need more documentation like this across the repo.

A few questions:
* What happens if PreparePendingWork() is not called? Is some response message sent automatically and synchronously?
* After releasing the CommandHandlerAsyncHandle, will the response message automatically be sent later?
* Is this schema only for sending StatusReport messages? Can it be used for response messages (example - send QueryImageResponse after QueryImage OTA command)